### PR TITLE
endpoint to get GKE clusters

### LINKED ISSUE #8001: Create endpoint to get GKE clusters for import
Create V2 endpoint:
GET /providers/gcp/clusters

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: endpoint to get GKE clusters

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8001

```release-note
NONE
```

### CODE REVIEW

File: pkg/handler/common/provider/gcp.go [hunk 1]
```diff
@@ -353,3 +355,21 @@ func filterGCPByQuota(instances apiv1.GCPMachineSizeList, quota kubermaticv1.Mac
...
 }
+
+func ListGKEClusters(ctx context.Context, sa string) (apiv2.GKEClusterList, error) {
+	clusters := apiv2.GKEClusterList{}
+	svc, project, err := gcp.ConnectToContainerService(sa)
+	if err != nil {
+		return clusters, err
+	}
+
+	req := svc.Projects.Zones.Clusters.List(project, "-")
```
Comment (lsviben) [inline pkg/handler/common/provider/gcp.go]: "-" means all zones? 

Could you make a const `allZones` so its clear?

File: pkg/provider/cloud/gcp/provider.go [hunk 2]
```diff
@@ -203,6 +204,36 @@ func ConnectToComputeService(serviceAccount string) (*compute.Service, string, e
...
+	projectID := sam["project_id"]
+	if projectID == "" {
+		return nil, "", errors.New("empty project_id")
+	}
+	conf, err := google.JWTConfigFromJSON(b, container.CloudPlatformScope)
+	if err != nil {
+		return nil, "", err
+	}
+	ctx := context.Background()
+	client := conf.Client(ctx)
```
Comment (lsviben) [inline pkg/provider/cloud/gcp/provider.go]: NIT: this part is the same in ConnectToComputeService, maybe we can extract it?

File: pkg/handler/v2/provider/gcp.go [hunk 3]
```diff
@@ -140,3 +142,62 @@ func GCPSubnetworkWithClusterCredentialsEndpoint(projectProvider provider.Projec
 		return providercommon.GCPSubnetworkWithClusterCredentialsEndpoint(ctx, userInfoGetter, projectProvider, privilegedProjectProvider, seedsGetter, req.ProjectID, req.ClusterID, req.Network)
 	}
 }
+
+// GKECommonReq represent a request with common parameters for GKE.
+type GKECommonReq struct {
+	// in: header
+	// name: ServiceAccount
+	ServiceAccount string
```
Comment (lsviben) [inline pkg/handler/v2/provider/gcp.go:151]: Could you add comments here and for Credential about how either one or the other can be used?

**Review by lsviben — COMMENTED** [review]
generally lgtm, some minor comments

File: pkg/handler/v2/provider/gcp.go (hunk 3 above)
Comment (zreigz) [reply pkg/handler/v2/provider/gcp.go:151]: done

**Review by lsviben — COMMENTED** [review]
/approve
/lgtm

**Review by mfranczy — APPROVED** [review]
/lgtm
/approve

**Pull request merged**